otter/integration/tests/test_convergence.py

```diff
@@ -0,0 +1,212 @@
+"""Tests covering foreseen or known edge cases for the Convergence subsystem.
+Tests covering self-healing should be placed in a separate test file.
+"""
+
+import os
+
+import treq
+
+from twisted.internet import reactor
+from twisted.internet.defer import gatherResults
+from twisted.internet.task import deferLater
+from twisted.internet.tcp import Client
+from twisted.trial import unittest
+from twisted.web.client import HTTPConnectionPool
+
+from otter import auth
+from otter.integration.lib.autoscale import (
+    BreakLoopException,
+    ScalingGroup,
+    ScalingPolicy,
+)
+from otter.integration.lib.identity import IdentityV2
+from otter.integration.lib.resources import TestResources
+
+from otter.util.deferredutils import retry_and_timeout
+from otter.util.http import check_success, headers
+from otter.util.retry import (
+    TransientRetryError,
+    repeating_interval,
+    transient_errors_except,
+)
+
+
+username = os.environ['AS_USERNAME']
+password = os.environ['AS_PASSWORD']
+endpoint = os.environ['AS_IDENTITY']
+flavor_ref = os.environ['AS_FLAVOR_REF']
+image_ref = os.environ['AS_IMAGE_REF']
+region = os.environ['AS_REGION']
+
+
+def find_end_points(rcs):
```

Comment:
This is not convergence specific and should probably belong in `otter.integration.lib.identity`. Also, non-blocking: is it possible for `self.identity.authenticate_user` to return `TestResources` instead of passing it and mutating it in this function?